tests/cpp/test_spmat_coo.cc

```diff
@@ -477,3 +504,67 @@ TEST(SpmatTest, COOGetDataAndIndices) {
   _TestCOOGetDataAndIndices<int32_t>();
   _TestCOOGetDataAndIndices<int64_t>();
 }
+
+template <typename IDX>
+void _TestCOOToCSRalgs() {
+  // Compare results between different CPU COOToCSR implementations.
+  // NNZ is chosen to be bigger than the limit for the "small" matrix algorithm.
+  // N is set to lay on border between "sparse" and "dense" algorithm choice.
+
+  const int64_t num_threads = std::min(256, omp_get_max_threads());
+  const int64_t min_num_threads = 3;
+
+  if (num_threads < min_num_threads) {
+    std::cerr << "[          ] [ INFO ]"
+              << "This test requires at least 3 OMP threads to work properly"
+              << std::endl;
+    GTEST_SKIP();
+    return;
+  }
+
+  // Select N and NNZ for COO matrix in a way than depending on number of
+  // threads different algorithm will be used.
+  // See WhichCOOToCSR in src/array/cpu/spmat_op_impl_coo.cc for details
+  const int64_t type_scale = sizeof(IDX) >> 1;
+  const int64_t small = 50 * num_threads * type_scale * type_scale;
+  // NNZ should be bigger than limit for small matrix algorithm
+  const int64_t nnz = small + 1234;
+  // N is chosen to lay on sparse/dense border
+  const int64_t n = type_scale * nnz / num_threads;
+  const IDX rows_nad_cols = n + 1;  // should be bigger than sparse/dense border
+
+  // TODO Set seed to random value when gtest allows using --gtest_random_seed
```

Comment:
Why it is important to set the seed to random value in the test?

If this is a TODO that to be done, file a issue and update it as: TODO(#1234): Set.
If this just a reminder, then remove the todo, and update it as, "Note that, it is better to set seed to random value when gtest allows ..., such that "reason".

Reply:
It seems not to be so important in this case. I can imagine that some randomly generated inputs could bring an unforeseen situation and discover a potential bug, so I prefer to use randomly generated inputs in the tests.  But as googletest doesn't bring the possibility of reproducing the issue with a given seed - setting seed randomly is counterproductive now.